packages/process/src/common/run-in-terminal.ts

```diff
@@ -0,0 +1,152 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { injectable } from 'inversify';
+import { createShellCommandLine, BashQuotingFunctions, PowershellQuotingFunctions, CmdQuotingFunctions, ShellQuoting, ShellQuotedString } from '../common/shell-quoting';
+
+export interface ProcessInfo {
+    executable: string
+    arguments: string[]
+}
+
+export interface CommandLineOptions {
+    cwd: string, args: string[], env?: {
+        [key: string]: string | null
+    }
+}
+
+/**
+ * Create command lines ready to be sent to a shell's stdin for evaluation.
+ */
+@injectable()
+export class RunInTerminalUtility {
```

Comment:
Also looking at the class it is not clear what it has to do with terminals? This class does not run anything in terminal. Maybe `ProcessCommandBuilder.build` instead? then it is relevant to the process and clear what it does. Or can we just inline it in `TerminalWidget`? Where else it supposed to be used?

Reply:
"Run In Terminal" as in running things in a terminal? And the comment explaining what it does? Will rename to "ShellCommandBuilder".